### PR TITLE
[Augmentation] Implement stats for various talents

### PR DESCRIPTION
**Pull request by Krealle**
### Description

@ljosberinn and Saeldur wanted stats so here they are.

Added stats for the following talents:
- Anachronism
- Pupil of Alexstrasza
- Volcanism
- Tectonic Locus
- Blistering Scales
  - Reactive Hide
  - Regenerative Chitlin
- Accretion
- Symbiotic Bloom

Also done a little house keeping:
- Learned how to spell Extension 
- Found another edgecase that needed fixing (Ebon Might not being extended when it should be)
- Implement Emerald Blossom extension for Sands of Time (When using Dream of Springs talent)

### Testing

**Symbiotic Bloom**
`/report/2qCQkDV9h8FKLy1r/18-Mythic+Rashok,+the+Elder+-+Kill+(6:17)/Genevoker/standard/statistics`
**Regenerative Chitlin & Reactive Hide**
`/report/8Lyzj4dghbn6YpZD/5-Mythic++Freehold+-+Kill+(28:06)/Tabalugatv/standard/statistics`

###  Screenshot(s):
image
image
image

### CODE REVIEW

File: src/analysis/retail/evoker/augmentation/modules/talents/Accretion.tsx [hunk 1]
```diff
@@ -22,10 +61,170 @@ class Accretion extends Analyzer {
...
+    this.shiftingSandsPercentage = (this.accretionShiftingSands / this.totalDamageDone) * 100;
+
+    const cdrUpheavelExtention =
+      (this.ebonMightUpheavelExtention / this.totalUpheavelCasts) * additionalUpheavalCastsViaCdr;
+
+    this.accretionEbonMight = (this.totalEbonMightDamage / EbonMightUptime) * cdrUpheavelExtention;
+
+    this.ebonMightPercentage = (this.accretionEbonMight / this.totalDamageDone) * 100;
+
+    if (debug) {
```
Comment (ljosberinn) [inline src/analysis/retail/evoker/augmentation/modules/talents/Accretion.tsx]: can / should this be removed?

File: src/analysis/retail/evoker/augmentation/modules/talents/Accretion.tsx [hunk 2]
```diff
@@ -22,10 +61,170 @@ class Accretion extends Analyzer {
...
+      );
+    }
+  }
+
+  statistic() {
+    // If the talent didn't provide enough CDR to actually be relevant
+    // Then no reason to show it
+    if (this.accretionShiftingSands + this.accretionEbonMight + this.accretionUpheavel === 0) {
+      return null;
+    }
```
Comment (ljosberinn) [inline src/analysis/retail/evoker/augmentation/modules/talents/Accretion.tsx]: I'm not sure this is how we should display it; isnt "dont play this talent on a fight this short" useful feedback? but also its fairly minor at the same time

File: src/analysis/retail/evoker/augmentation/modules/talents/BlisteringScales.tsx [hunk 3]
```diff
@@ -0,0 +1,161 @@
...
+import STATISTIC_ORDER from 'parser/ui/STATISTIC_ORDER';
+import TalentSpellText from 'parser/ui/TalentSpellText';
+import DonutChart from 'parser/ui/DonutChart';
+import { SpellLink } from 'interface';
+
+/**
+ * Blistering Scales is essentially Augmentations external
+ * it provides your target with 30% of your armor along with doing AoE
+ * to nearby enemies when said target is meleed. Consuming 1 stack.
+ * This effect has an ICD of 3-4s
```
Comment (ljosberinn) [inline src/analysis/retail/evoker/augmentation/modules/talents/BlisteringScales.tsx]: irrelevant but it's lower, over 27s you can end up at 2-3 stacks left (27/13 => ~2s ICD)

File: src/analysis/retail/evoker/augmentation/modules/talents/BlisteringScales.tsx [hunk 4]
```diff
@@ -0,0 +1,161 @@
...
+        </TalentSpellText>
+
+        {this.hasReactiveHide && (
+          <TalentSpellText talent={TALENTS.REACTIVE_HIDE_TALENT}>
+            <ItemDamageDone amount={this.reactiveHideDamage} />
+          </TalentSpellText>
+        )}
+
+        {this.hasRegenerativeChitin && (
+          <div className="pad">
```
Comment (ljosberinn) [inline src/analysis/retail/evoker/augmentation/modules/talents/BlisteringScales.tsx:151]: there's pad here, but not for reactive hide, intentional? (havent checked it out locally)

Comment (Krealle) [reply src/analysis/retail/evoker/augmentation/modules/talents/BlisteringScales.tsx:151]: the pad class is needed to make the donut behave properly, and only chitin has a donut

File: src/analysis/retail/evoker/augmentation/modules/talents/Accretion.tsx (hunk 2 above)
Comment (Krealle) [reply src/analysis/retail/evoker/augmentation/modules/talents/Accretion.tsx]: oh yea that's prolly better, will change that up

File: src/analysis/retail/evoker/augmentation/modules/talents/BlisteringScales.tsx (hunk 3 above)
Comment (Krealle) [reply src/analysis/retail/evoker/augmentation/modules/talents/BlisteringScales.tsx]: will be amended

File: src/analysis/retail/evoker/augmentation/modules/talents/Accretion.tsx (hunk 1 above)
Comment (Krealle) [reply src/analysis/retail/evoker/augmentation/modules/talents/Accretion.tsx]: You thinking about the pencentages or the debugging ?

Comment (ljosberinn) [reply src/analysis/retail/evoker/augmentation/modules/talents/Accretion.tsx]: debugging

File: src/analysis/retail/evoker/augmentation/modules/talents/Accretion.tsx (hunk 2 above)
Comment (Krealle) [reply src/analysis/retail/evoker/augmentation/modules/talents/Accretion.tsx]: image

File: src/analysis/retail/evoker/augmentation/modules/talents/Accretion.tsx [hunk 5]
```diff
@@ -170,17 +165,40 @@ class Accretion extends Analyzer {
...
+              />
+            </TalentSpellText>
+            <div className="pad">
+              <label>
+                <Trans>Damage sources</Trans>
+              </label>
+              <DonutChart items={damageSources} />
+            </div>
+          </div>
+        )) || (
```
Comment (ljosberinn) [inline src/analysis/retail/evoker/augmentation/modules/talents/Accretion.tsx]: minor but the better approach here would be ? : instead of && ||

**Comment by ljosberinn** [comment]
send it

**Pull request merged**